cmd/query/app/http_handler.go

```diff
@@ -193,6 +196,43 @@ func (aH *APIHandler) getOperationsLegacy(w http.ResponseWriter, r *http.Request
 	aH.writeJSON(w, r, &structuredRes)
 }
 
+func (aH *APIHandler) transformOTLP(w http.ResponseWriter, r *http.Request) {
+	body, err := ioutil.ReadAll(r.Body)
+	if aH.handleError(w, err, http.StatusBadRequest) {
+		return
+	}
+
+	var uiErrors []structuredError
+	batches, err := apiv3.OTLP2model(body)
```

Comment:
there is no need for those functions to be defined in apiv3 package, you can create a similar `transform_otlp.go` in this package.